# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/mlgpu/CanvasLayerMLGPU.h
# Commit: 204b899f436d
# Full Hash: 204b899f436d99f0eae759af3ccfcbaabfbbb428
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 09:34:54
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 3f5baab34b1f -r 204b899f436d gfx/layers/mlgpu/CanvasLayerMLGPU.h
--- a/gfx/layers/mlgpu/CanvasLayerMLGPU.h	Thu Jun 11 06:36:55 2020 +0000
+++ b/gfx/layers/mlgpu/CanvasLayerMLGPU.h	Thu Jun 11 06:37:35 2020 +0000
@@ -42,7 +42,7 @@
   MOZ_LAYER_DECL_NAME("CanvasLayerMLGPU", TYPE_CANVAS)
 
  protected:
-  CanvasRenderer* CreateCanvasRendererInternal() override {
+  RefPtr<CanvasRenderer> CreateCanvasRendererInternal() override {
     MOZ_CRASH("Incompatible surface type");
     return nullptr;
   }